projects/ngx-charts-on-fhir/src/lib/fhir-data/fhir-data.service.ts

```diff
@@ -95,15 +98,64 @@ export class FhirDataService {
       // use maximum page size on every request to improve performance
       url = addCountParam(url);
       const request = currentPatientOnly ? this.client.patient.request.bind(this.client.patient) : this.client.request.bind(this.client);
+
       request(url, { pageLimit: 0, onPage })
         .then(() => subscriber.complete())
         .catch((error) => subscriber.error(error));
       return teardownLogic;
-    });
+    }).pipe(
+      retryBackoff({
+        initialInterval: this.INITIAL_INTERVAL,
+        maxRetries: this.MAX_RETRIES,
+        resetOnSuccess: true,
+        shouldRetry: (error) => {
+          return this.errorStatusCheck(error); // always retry
+        },
+      })
+    );
   }
 
-  addPatientData(resource: Resource) {
-    return this.client?.create(resource);
+  errorStatusCheck(error: any) {
+    switch (error.status) {
+      case 500:
+        return true;
+      case 501:
+        return true;
+      case 502:
+        return true;
+      case 503:
+        return true;
+      case 504:
+        return true;
+      default:
+        return false;
+    }
+  }
+
+  addPatientData<T extends Resource>(resource: T): Observable<T> {
```

Comment:
For consistency, can we use `R` as the type param name so it matches `getPatientData`?